src/docformer/modeling.py

```diff
@@ -54,44 +52,37 @@ def forward(self, x):
         x = rearrange(x, "b e s -> b s e")  # b -> batch, e -> embedding dim, s -> sequence length
         return x
 
-
 class DocFormerEmbeddings(nn.Module):
     """Construct the embeddings from word, position and token_type embeddings."""
 
     def __init__(self, config):
         super(DocFormerEmbeddings, self).__init__()
 
         self.config = config
-        self.word_embeddings = nn.Embedding(
-            config["vocab_size"],
-            config["hidden_size"],
-            padding_idx=config["pad_token_id"],
-        )
 
         self.position_embeddings_v = PositionalEncoding(
             d_model=config["hidden_size"],
             dropout=0.1,
             max_len=config["max_position_embeddings"],
         )
 
-        self.x_v = nn.Embedding(config["max_2d_position_embeddings"], config["coordinate_size"])
         self.x_topleft_position_embeddings_v = nn.Embedding(config["max_2d_position_embeddings"], config["coordinate_size"])
         self.x_bottomright_position_embeddings_v = nn.Embedding(config["max_2d_position_embeddings"], config["coordinate_size"])
         self.w_position_embeddings_v = nn.Embedding(config["max_2d_position_embeddings"], config["shape_size"])
-        self.x_topleft_distance_to_prev_embeddings_v = nn.Embedding(config["max_2d_position_embeddings"], config["shape_size"])
-        self.x_bottomleft_distance_to_prev_embeddings_v = nn.Embedding(config["max_2d_position_embeddings"], config["shape_size"])
-        self.x_topright_distance_to_prev_embeddings_v = nn.Embedding(config["max_2d_position_embeddings"], config["shape_size"])
-        self.x_bottomright_distance_to_prev_embeddings_v = nn.Embedding(config["max_2d_position_embeddings"], config["shape_size"])
-        self.x_centroid_distance_to_prev_embeddings_v = nn.Embedding(config["max_2d_position_embeddings"], config["shape_size"])
+        self.x_topleft_distance_to_prev_embeddings_v = nn.Embedding(2*config["max_2d_position_embeddings"] + 1, config["shape_size"])
```

Comment:
What's the reason for going for `2*config["max_2d_position_embeddings"] + 1` instead of the original?

Reply:
The reason for this is mentioned below:
In the relative values, we are not taking the absolute values (following the conventions used by other algorithms such as LayoutLM, LayoutLMV2), instead what we do is:
1. Clamp the values ( let us say between -1024,1024) #1024 is the max_2d_position_embeddings 
2. Add a bias of 1024, so that now the values ranges between (0,2048)
3. Hence for the embedding layer, which are encoding the relative distance features, should havs a embedding layer range (0,2048, but I added 1, because of just remaining on the safe side

For the clamping, we have two options:
1. Either do it in the Extract Features of modeling.py
2. Or do it while making the dataset class

Currently, I thought of doing it in Dataset class, but maybe we can change it and write it in modeling.py file